packages/storybook/integration/helper-fn.ts

```diff
@@ -20,15 +24,33 @@ export const IMAGE_SNAPSHOT_CONFIG = {
   },
 } as MatchImageSnapshotOptions;
 
+let browserInstance;
 export async function startBrowser() {
```

Comment:
I had a look at the helpers I wrote in the past to write tests with puppeteer, if that help:

Usage, **createContext()**
```jsx
import config from 'config'
import { ACCOUNT_STATUSES, SCOPES } from 'api/constants'
import { createContext } from 'test/e2e/setup'
import signIn from 'test/e2e/modules/signIn'
import matchers from 'test/e2e/modules/matchers'
import screenshot from 'test/e2e/modules/screenshot'

describe('signIn', () => {
  const context = createContext()

  it('should work', async () => {
    const account = await context.factory.create(
      'Account',
      { status: ACCOUNT_STATUSES.APPROVED },
      { scopes: [SCOPES.CONTRIBUTOR] }
    )
    const { page, i18n } = context

    await page.goto(`${config.get('contributor.url')}/sign-in`)
    await matchers.toMatch(page, i18n.t('contributorSignIn:apply'), {
      timeout: 'navigation',
    })
    await screenshot(page, 'contributor-sign-in')
    await page.type('input[name="email"]', account.email)
    await page.type('input[name="password"]', account.password)
    await page.type('input[name="password"]', '\u000d')
    await matchers.toMatchUrl(page, `${config.get('contributor.url')}/`)
    await matchers.toMatch(page, i18n.t('contributorDashboard:imageRequiresEditing'), {
      timeout: 'navigation',
    })
    await page.waitFor('[data-e2e="appBar.user.menu"]')
    await screenshot(page, 'contributor-dashboard')
    await page.click('[data-e2e="appBar.user.menu"]')
    await matchers.toMatchElement(page, '[role="menu"] > div:first-child', {
      text: `${account.firstName} ${account.lastName}`,
    })
  })

  it('should redirect when trying to access the sign in page', async () => {
    const account = await context.factory.create(
      'Account',
      { status: ACCOUNT_STATUSES.APPROVED },
      { scopes: [SCOPES.CONTRIBUTOR] }
    )
    await signIn({ context, account, scope: SCOPES.CONTRIBUTOR })

    const { page } = context
    await page.goto(`${config.get('contributor.url')}/settings/profile`)
    expect(page.url()).toBe(`${config.get('contributor.url')}/settings/profile`)
    await page.goto(`${config.get('contributor.url')}/sign-in`)
    expect(page.url()).toBe(`${config.get('contributor.url')}/`)
  })
})

```

**test/e2e/setup.js**
```jsx
/* eslint-disable no-underscore-dangle, import/prefer-default-export, import/no-extraneous-dependencies */
/* global jasmine */

import fetch from 'isomorphic-fetch'
import knexfile from 'knexfile'
import puppeteer from 'puppeteer'
import log from 'modules/scripts/log'
import waitUntil from 'modules/async/waitUntil'
import hackTranslate from 'modules/i18n/hackTranslate'
import i18nInit from 'web/modules/i18n/server'
import getFactory from 'test/modules/api/factory'
import createKnex from 'api/services/database/createKnex'
import truncateAll from 'api/services/database/truncateAll'
import { IMAGE_STATUSES } from 'api/constants'

jasmine.DEFAULT_TIMEOUT_INTERVAL = 6000e3

async function getBrowserWSEndpoint() {
  if (process.env.__BROWSER_WS_ENDPOINT__) {
    return process.env.__BROWSER_WS_ENDPOINT__
  }

  const browserRemoteUrl = process.env.BROWSER_REMOTE_URL

  if (!browserRemoteUrl) {
    throw new Error('process.env.BROWSER_REMOTE_URL is missing')
  }

  const browserWSEndpoint = await waitUntil(async () => {
    log.info({
      name: 'browser',
      msg: `Fetch data from ${browserRemoteUrl}`,
      force: true,
    })
    let version

    try {
      version = await fetch(`${browserRemoteUrl}/json/version`, {
        method: 'GET',
      }).then(result => result.json())
    } catch (err) {
      // ...
    }

    return {
      predicate: version,
      result: version ? version.webSocketDebuggerUrl : null,
    }
  })

  log.info({
    name: 'browser',
    msg: `Resolved browser endpoint ${browserWSEndpoint} `,
    force: true,
  })

  process.env.__BROWSER_WS_ENDPOINT__ = browserWSEndpoint

  return browserWSEndpoint
}

async function launchBrowser() {
  const browserWSEndpoint = await getBrowserWSEndpoint()
  return puppeteer.connect({ browserWSEndpoint, slowMo: 0 })
}

function handleError(err) {
  throw err
}

const knexConfig = knexfile[process.env.NODE_ENV]

export function createContext() {
  const context = {}

  const screenSize = process.env.SCREEN_SIZE
  if (!screenSize) {
    throw new Error('Missing process.env.SCREEN_SIZE')
  }
  const [width, height] = screenSize.split('x').map(Number)

  beforeAll(async () => {
    // console.time('db')
    context.knex = createKnex({ connection: { database: knexConfig.connection.database } })
    context.factory = getFactory(context.knex.models)
    // console.timeEnd('db')

    await Promise.all([
      truncateAll(context.knex),

      (async () => {
        // console.time('browser')
        context.browser = await launchBrowser()
        const pages = await context.browser.pages()
        if (!pages.length) {
          pages.push(await context.browser.newPage())
        }
        context.page = pages[0]
        context.page.setDefaultNavigationTimeout(
          process.env.NODE_ENV === 'production' ? 10e3 : 60e3
        )
        context.page.addListener('pageerror', handleError)
        await context.page.setViewport({ width, height: height - 83 })
        // console.timeEnd('browser')
      })(),
      (async () => {
        // console.time('i18n')
        const i18n = await i18nInit()
        context.i18n = hackTranslate(i18n)
        // console.timeEnd('i18n')
      })(),
    ])
  })

  // Avoid duplicating beforeAll truncate
  let truncateDo = false
  beforeEach(async () => {
    // Kill all the in-flight requests
    const html =
      '<!DOCTYPE html><html><head></head><body>beforeEach, kill all the pending requests</body></html>'
    await context.page.goto(`data:text/html,${html}`)
    await Promise.all([
      (async () => {
        if (truncateDo) {
          await truncateAll(context.knex)
        }
        truncateDo = true
      })(),
      context.page._client.send('Network.clearBrowserCookies'),
    ])

    await context.factory.create('Image', {
      customerHomeBackground: true,
      status: IMAGE_STATUSES.APPROVED_ONLINE,
    })
  })

  afterAll(async () => {
    context.page.removeListener('pageerror', handleError)
    await Promise.all([context.browser.disconnect(), context.knex.destroy()])
  })

  return context
}
```